### PR TITLE
Added map_boundary_to_manifold_ids.

### PR DESCRIPTION
**Pull request by luca-heltai**
@jppelteret , is this similar to what you had in mind?

I'm not sure about the container types. This could be done also with a 

```
std::map<types::boundary_id, types::manifold_id> boundary_to_manifold_ids;
```
and an optional
```
std::map<types::boundary_id, types::boundary_id> new_boundary_ids;
```
But this seemed to be a little easier. What do people think?

### CODE REVIEW

File: source/grid/grid_tools.cc [hunk 1]
```diff
@@ -3257,24 +3289,31 @@ namespace GridTools
...
               }
+
+            if (dim >= 3)
+              for (signed int e=0; e<static_cast<signed int>(GeometryInfo<dim>::lines_per_face); ++e)
+                {
+                  auto bid = cell->face(f)->line(e)->boundary_id();
+                  auto ind = std::find(src_boundary_ids.begin(), src_boundary_ids.end(), bid)-
+                             src_boundary_ids.begin();
+                  if ((unsigned int)ind < src_boundary_ids.size())
+                    cell->face(f)->line(e)->set_boundary_id(reset_boundary_ids[ind]);
```
Comment (jppelteret) [inline source/grid/grid_tools.cc]: Am I correct that for an edge shared by faces with different boundary id's then its not really possible to predetermine what the resulting ID on the edge will be? If so then maybe its best if we document this.

Comment (luca-heltai) [reply source/grid/grid_tools.cc]: Mmmmh. Why would it be so? The edge has its own boundary id, independent on the face. Here I'm resetting it if it is in the list of ids to reset...

Comment (jppelteret) [reply source/grid/grid_tools.cc]: Yes, of course. My mistake!

File: include/deal.II/grid/grid_tools.h [hunk 2]
```diff
@@ -2269,6 +2269,35 @@ namespace GridTools
   void copy_boundary_to_manifold_id(Triangulation<dim, spacedim> &tria,
                                     const bool reset_boundary_ids=false);
 
+  /**
+   * Map the given boundary ids to the given manifold ids on faces and
+   * edges at the boundary.
+   *
+   * This function copies the boundary ids of the boundary faces and
+   * edges that are present in the parameter @p src_boundary_ids to
+   * the corresponding manifold_id in @p dst_manifold_ids, of the same
```
Comment (jppelteret) [inline include/deal.II/grid/grid_tools.h]: corresponding manifold_id in @p dst_manifold_ids
->
corresponding manifold id in @p dst_manifold_ids

File: source/grid/grid_tools.cc [hunk 3]
```diff
@@ -3257,24 +3289,31 @@ namespace GridTools
...
               }
+
+            if (dim >= 3)
+              for (unsigned int e=0; e<GeometryInfo<dim>::lines_per_face; ++e)
+                {
+                  const auto bid = cell->face(f)->line(e)->boundary_id();
+                  const auto ind = std::find(src_boundary_ids.begin(), src_boundary_ids.end(), bid)-
+                                   src_boundary_ids.begin();
+                  if ((unsigned int)ind < src_boundary_ids.size())
+                    cell->face(f)->line(e)->set_boundary_id(reset_boundary_ids[ind]);
```
Comment (jppelteret) [inline source/grid/grid_tools.cc]: @luca-heltai Do we also perhaps need to set the edge manifold id here? Its just struck me now that the call to `cell->face(f)->set_manifold_id()` [does not propagate](https://dealii.org/8.5.0/doxygen/deal.II/group__manifold.html#gac23367ec88e9282815b9a03df8af5431) this information to the edges.

Comment (jppelteret) [reply source/grid/grid_tools.cc]: Well, I've convinced the setting of edge manifold id's works as expected, so my question can be ignored. I hope you don't mind that I added my test specifically for this purpose.

Comment (luca-heltai) [reply source/grid/grid_tools.cc]: :)

Comment (bangerth) [reply source/grid/grid_tools.cc]: btw, instead of doing it by hand for edges, you can call `set_all_boundary_ids()`.

Comment (luca-heltai) [reply source/grid/grid_tools.cc]: Nope. You need to make sure that the original id was the one you want to replace. If you use `set_all_boundary_ids`, this  is done regardless of the original id.

**Pull request merged**